src/frontend/src/helpers/Storage/LocalStorage.ts

```diff
@@ -153,104 +108,108 @@ class Store implements StorageInterface {
-        await this.db.changelog.add({
-          file: {
-            project: proj, 
-            file: name, 
-            checksum: current.checksum
-          },
-          type: "deleteFile"
-        });
-      }
-      await this.db.files.delete(key);
+      let current = await this.db.files.get(id);
+      await this.db.changelog.add({
+        file: {
+          project: proj,
+          file: name,
+          checksum: current.checksum
+        },
+        type: "deleteFile"
+      });
+      await this.db.files.delete(id);
       // also remove from run files
       let dbProj = await this.db.projects.get(proj);
       for (const q in dbProj.runs) {
-        if (dbProj.runs[q] == name) {
+        if (R.equals(dbProj.runs[q] == id)) {
           delete dbProj.runs[q];
         }
       }
       await this.db.projects.update(proj, dbProj)
     });
   }
 
-  public async renameFile(proj: string, name: string, newName: string, checksum: string): Promise<string> {
-    const key: FileID = [proj, name];
+  public async renameFile(id: FileID, newName: string): Promise<void> {
+    const proj = id[0];
+    const name = id[1];
     const tbs = [this.db.files, this.db.projects, this.db.changelog];
     return this.db.transaction('rw', tbs, async () => {
-      let result = await this.db.files.get(key);
+      let result = await this.db.files.get(id);
       result.name = newName;
       await this.db.files.add(result);
       // also rename in run files
       let dbProj = await this.db.projects.get(proj);
       for (const q in dbProj.runs) {
-        if (dbProj.runs[q] == name) {
-          dbProj.runs[q] = newName;
+        if (dbProj.runs[q] == id) {
+          dbProj.runs[q] = [proj, newName];
         }
       }
-      await this.db.projects.update(proj, dbProj)
-      if (! checksum) {
-        var change = {
-          file:{
-            project: proj, 
-            name: newName
-          },
-          type: "newFile",
-          contents: result.contents,
-          history: result.history
-        }
-        await this.db.changelog.add(change);
+      await this.db.projects.update(proj, dbProj);
+      var change = {
+        file:{
+          project: proj,
+          name: newName
+        },
+        type: "newFile",
+        contents: result.contents
       }
-      await this.deleteFile(proj, name, !!checksum);
-      return result.checksum;
+      await this.db.changelog.add(change);
+      await this.deleteFile(id);
     });
   }
 
-  public async getFileToRun(proj: string, question: string): Promise<string|undefined> {
+  public async getRunFile(proj: ProjectID, question: string): Promise<FileID|undefined> {
     let p = await this.db.projects.get(proj);
     return p.runs[question];
   }
 
   // a file name is (test|q*|common)/name
-  public async setFileToRun(proj: string, question: string, file: string): Promise<void> {
+  public async setRunFile(file: FileID, question: string): Promise<void> {
```

Comment:
Avoid changing the contract -- Also, you still _need_ to pass in the project, as in the future FileID won't store the project information.